client/my-sites/customer-home/main.jsx

```diff
@@ -624,7 +624,8 @@ const connectHome = connect(
 		const createdAt = getSiteOption( state, siteId, 'created_at' );
 
 		return {
-			displayChecklist: isEligibleForDotcomChecklist( state, siteId ),
+			displayChecklist:
+				isEligibleForDotcomChecklist( state, siteId ) && hasChecklistData && ! isChecklistComplete,
```

Comment:
Should we update the selector instead?

Reply:
We should audit first existing usages and make sure that wouldn't cause any side effect: https://github.com/Automattic/wp-calypso/search?q=isEligibleForDotcomChecklist&unscoped_q=isEligibleForDotcomChecklist

Code using `isEligibleForDotcomChecklist` might expect the selector to return true if site is eligible for the checklist despite being completed.